include/dynd/kernels/string_concat_kernel.hpp

```diff
@@ -42,7 +22,7 @@ namespace dynd {
         dynd::string *d = reinterpret_cast<dynd::string *>(dst);
         const dynd::string *const *s = reinterpret_cast<const dynd::string *const *>(src);
 
-        detail::concat_one_string(2, d, s);
+        string_concat(2, d, s);
       }
     };
 
```

Comment:
I'd probably just call `string_concat(2, reinterpret_cast<dynd::string *>(dst), reinterpret_cast<string *>(src[0]));` and skip the temporaries, but it's up to you.